### PR TITLE
Fix deployment (in and out of docker)

### PR DESCRIPTION
**Pull request by Trolldemorted**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,7 +13,8 @@ services:
       - ROCKET_ENV=production
       - ROCKET_LOG=normal
       - ROCKET_SECRET_KEY=whs/vijJnEoWN9Xgf25oJDn2yUtvsNuhm0eMNxZe6CI=
-      - SERVER_ADMIN=root@vulnbox,root@vmhost
+      - SERVER_ADMIN=benni@benni.stronk.pw:8022
```
Comment (fsck) [inline docker-compose.yml]: I feel like this field should be filled with clear example hostnames or (preferably) we should add some testing code for launching a local sshd and then point to that.

File: src/deploy.rs [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
-        let mut content = vec!();
-        File::open(file_name)?.read_to_end(&mut content)?;
-        let tcp = TcpStream::connect(&destination.address)?;
-        let mut sess = Session::new().unwrap();
-        sess.handshake(&tcp)?;
-        sess.userauth_agent(&destination.userauth_agent)?;
-        let mut remote_file = sess.scp_send(Path::new("~/.ssh/authorized_keys"), 0o644, content.len() as u64, None)?;
-        remote_file.write_all(&content)?;
+        Command::new("scp")
+            .args(&["-i", "./data/id_ed25519",
```
Comment (fsck) [inline src/deploy.rs:10]: I am not very happy with this solution but if you insist you need ed25519 keys and absolutely can not live with ecdsa keys than I will endure this. But maybe we can add some more differentiated error handling.

File: src/storage.rs [hunk 3]
```diff
@@ -89,3 +95,14 @@ pub fn generate_authorized_key_files(destinations: &Vec<Destination>) -> Result<
...
     Ok(())
 }
+
+pub fn load_deploy_keypair() -> Result<(), EnokeysError> {
+    generate_deploy_keypair()
+}
+
+fn generate_deploy_keypair() -> Result<(), EnokeysError> {
+    Command::new("ssh-keygen")
+        .args(&["-t", "ed25519", "-m", "pem", "-f", "./data/id_ed25519"])
```
Comment (fsck) [inline src/storage.rs:105]: This should be configurable options (which key[path] to use; whether it should be generate at startup, if not found)

File: src/storage.rs [hunk 4]
```diff
@@ -89,3 +95,14 @@ pub fn generate_authorized_key_files(destinations: &Vec<Destination>) -> Result<
     }
     Ok(())
 }
+
+pub fn load_deploy_keypair() -> Result<(), EnokeysError> {
```
Comment (fsck) [inline src/storage.rs:99]: This function is a stub.

**Review by fsck — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (Trolldemorted) [reply docker-compose.yml]: Imho we shouldn't have a docker-compose.yml in this repository at all - an example yaml in the readme should be enough.

File: src/deploy.rs (hunk 2 above)
Comment (Trolldemorted) [reply src/deploy.rs:10]: You are welcome to implement deployment with ssh2 - I just kept it as simple as possible for the MVP.

File: src/storage.rs (hunk 3 above)
Comment (Trolldemorted) [reply src/storage.rs:105]: Is that within the scope of this PR?

File: src/storage.rs (hunk 4 above)
Comment (Trolldemorted) [reply src/storage.rs:99]: Obviously - I kept it here until a deploy key management decision is there.

**Comment by fsck** [comment]
Rolled into #28

**Pull request closed without merging**